Add FIPS compliance to agents plugin

stafot opened the pull request:
# Add FIPS Compliance to Agents Plugin

## Summary

This PR adds Federal Information Processing Standards (FIPS) compliance support to the Mattermost Agents Plugin, enabling the plugin to be deployed in FIPS-compliant environments such as government agencies, financial institutions, and other regulated industries.

## 🎯 What's Changed

### Core FIPS Support
- **New FIPS Build Target**: Added `make dist-fips` command to build FIPS-compliant plugin distributions
- **Docker-based FIPS Builds**: Uses official Mattermost FIPS-compliant Go image (`cgr.dev/mattermost.com/go-msft-fips:1.24.4`) for building
- **Dual Distribution Support**: Plugin now builds both standard and FIPS-compliant versions simultaneously

### Build System Enhancements
- **Updated Go Version**: Upgraded from Go 1.23 to Go 1.24.0 for better FIPS compliance
- **Enhanced Makefile**: Added comprehensive FIPS build targets and helper functions
- **Improved CI/CD**: Updated GitHub Actions to build and distribute both plugin versions

### CI/CD Improvements
- **Dependabot Integration**: Added automated dependency updates for GitHub Actions
- **Enhanced Build Pipeline**: CI now builds both normal and FIPS distributions
- **Artifact Management**: Improved artifact handling and retention policies
- **Security Hardening**: Pinned GitHub Actions to specific commit hashes for better security

## 🔧 Technical Details

### FIPS Build Process
The FIPS build process:
1. Uses the official Mattermost FIPS-compliant Go image
2. Builds server binaries with FIPS-compliant cryptographic libraries
3. Creates separate distribution packages with `-fips` suffix
4. Maintains compatibility with existing plugin deployment workflows

### New Make Targets
- `make dist-fips` - Builds FIPS-compliant plugin distribution
- `make dist-all` - Builds both standard and FIPS distributions
- `make server-fips` - Builds only the FIPS-compliant server binaries

### File Structure

dist/ # Standard plugin distribution
dist-fips/ # FIPS-compliant plugin distribution
├── plugin-linux-amd64-fips
└── plugin.json

## 🚀 Usage

### Building FIPS Plugin
```bash
# Build FIPS-compliant plugin
make dist-fips

# Build both versions
make dist-all
```

### Deployment
Both plugin versions are automatically built in CI and available as release artifacts. The FIPS version can be deployed to FIPS-compliant environments while maintaining the standard version for regular deployments.

## 🧪 Testing

- [x] FIPS build process tested locally
- [x] CI pipeline updated and tested
- [x] Both distributions build successfully
- [x] Plugin functionality verified in both versions

## 📋 Checklist

- [x] Add FIPS build support to Makefile
- [x] Update CI workflow for dual distribution builds
- [x] Pin GitHub Actions to specific versions
- [x] Add dependabot configuration
- [x] Update Go version to 1.24.0
- [x] Test FIPS build process
- [x] Verify plugin functionality in both distributions

## 🔒 Security & Compliance

This change enables the plugin to meet FIPS 140-2 compliance requirements by:
- Using FIPS-validated cryptographic modules
- Building with FIPS-compliant Go toolchain
- Maintaining separate build processes for compliance verification

## �� Related

- **Issues**: [CLD-9438](https://mattermost.atlassian.net/browse/CLD-9438), [CLD-9440](https://mattermost.atlassian.net/browse/CLD-9440)
- **Type**: Enhancement
- **Breaking Changes**: None
- **Migration**: No migration required - existing deployments continue to work unchanged

## 🎉 Impact

This enhancement significantly expands the plugin's deployment capabilities, making it suitable for:
- Government and military environments
- Financial services and healthcare organizations
- Any environment requiring FIPS compliance
- Enterprise customers with strict security requirements

The change maintains full backward compatibility while adding enterprise-grade compliance features.


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

* **Chores**
  * Enhanced CI/CD pipeline with FIPS-compliant build support alongside standard distributions
  * Improved artifact signing and verification in release process
  * Added automated dependency management via Dependabot
  * Strengthened workflow security with tighter permission controls

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

stafot committed:
WIP: Add FIPS compliance to agents plugin

Signed-off-by: Stavros Foteinopoulos <stafot@gmail.com>

stafot committed:
Fixes after testing

Signed-off-by: Stavros Foteinopoulos <stafot@gmail.com>

agarciamontoro reviewed (commented):
Nice! I left a couple of nits below, and I have a general suggestion that can actually be applied to the other plugins PRs: can we add a check like [the one proposed for the server](https://github.com/mattermost/mattermost/pull/33549/files#diff-3c42926021fd46d4374fa37b1bcf6e332fcb6fdd5023caf939d401aea430dbb8R166-R175) to verify the FIPS builds?
```sh
ifeq ($(FIPS_ENABLED),true)
	cp $(GOBIN_HOST)/$(MM_BIN_NAME) $(GOBIN_HOST)/$(MMCTL_BIN_NAME) $(DIST_PATH_GENERIC)/bin # from native bin dir, not cross-compiled
	@# FIPS verification checks
	@echo "Verifying FIPS build settings..."
	$(GO) version -m $(GOBIN_HOST)/$(MM_BIN_NAME) | grep -q "GOEXPERIMENT=systemcrypto" || (echo "ERROR: FIPS binary missing GOEXPERIMENT=systemcrypto" && exit 1)
	$(GO) version -m $(GOBIN_HOST)/$(MM_BIN_NAME) | grep -q "\-tags=requirefips" || (echo "ERROR: FIPS binary missing -tags=requirefips" && exit 1)
	@echo "Verifying OpenSSL integration..."
	$(GO) tool nm $(GOBIN_HOST)/$(MM_BIN_NAME) | grep -q "func_go_openssl_OpenSSL_version" || (echo "ERROR: FIPS binary missing OpenSSL integration" && exit 1)
	@echo "FIPS verification checks passed"
else
```

agarciamontoro reviewed (changes requested):
(I meant to Request changes, sorry)

stafot committed:
Review fixes

Signed-off-by: Stavros Foteinopoulos <stafot@gmail.com>

stafot committed:
add artifacts sign

Signed-off-by: Stavros Foteinopoulos <stafot@gmail.com>

stafot committed:
Remove sign from CI

Signed-off-by: Stavros Foteinopoulos <stafot@gmail.com>

stafot committed:
Add manual artifacts CI sign for plugins - fixes

Signed-off-by: Stavros Foteinopoulos <stafot@gmail.com>

stafot committed:
add version prefix

Signed-off-by: Stavros Foteinopoulos <stafot@gmail.com>

stafot committed:
fix version suffix

Signed-off-by: Stavros Foteinopoulos <stafot@gmail.com>

stafot committed:
fix version suffix 2

Signed-off-by: Stavros Foteinopoulos <stafot@gmail.com>

agarciamontoro committed:
Merge tag 'v1.4.0' into CLD-9438-build-agents-fips-complaint

Bumping mattermost-plugin-agents to Minor version 1.4.0

agarciamontoro committed:
Merge tag 'v1.6.0' into CLD-9438-build-agents-fips-complaint

Bumping mattermost-plugin-agents to Minor version 1.6.0

# -----BEGIN PGP SIGNATURE-----
#
# iQIzBAABCgAdFiEEPRP5d5L03rt/wvKse019lts08b8FAmkUzusACgkQe019lts0
# 8b9vDg/+I4JGs2tD7Spbi2bEvQ96hMn8CGiTl4Kf2lDryevWamOQ+pQpxERf2LAQ
# 10jPhGdkONodJ3v2YytS/z1y+rHXMeQXRZF+TeZ0xdCLS4qrFXnLjXz+L9l4U9Ju
# 8QykVKc5qGlkLK/BXZQt2yvjVGMW26wDTbXZRnfmCMzLfTDcNMmp5s5s3CySzZFW
# jU+4eVqqyIuQGKzVEq/3OBqKuDvhLXbe5psr/uFmyXjHv1atv3sJ7ggL77eU5BCi
# MbT8yvTPncynsFxyy1OQiG4DqOIwKHU8Ik8nJ9Ck/xQKHtgTM7ElJ0PlvZmYXNqm
# GCWPIp2wk7M34ln1uyPDUdWwPPbidJXnknovyKU1awV95QiW5uUoM8eJzqZuGguT
# NyOle5ipqJI2sXQ6+NdWXuRVEVyNW+LVz3m4qUodBAu7cSWeLStv3gVjB4jORYKU
# wOA7ay656ylS0YwNz+GA8EzZhqkYZnZNJDa6aD5cZibfu73EvHk8g05mJM0xg0Ty
# 5n3wADbEyDFizH2sptUDvILgCYyqkSdjnbddYMtdj5lgR0+Op/26LbrGOKgEvUe/
# zWw1R1uaSmfiLizVrEveTrXx6a3pjreXRV9EwFRXbYmjgEVh8XxHJIfw+0c0Hr4q
# FoTTgW8Nt+JXNrVKkv8t+ueuJav1JTwRbLzvrNdcIVZ7XMY0mao=
# =AhbM
# -----END PGP SIGNATURE-----
# gpg: Signature made mié 12 nov 2025 19:16:11 CET
# gpg:                using RSA key 3D13F97792F4DEBB7FC2F2AC7B4D7D96DB34F1BF
# gpg: Can't check signature: No public key

agarciamontoro committed:
Merge tag 'v1.6.1' into CLD-9438-build-agents-fips-complaint

Bumping mattermost-plugin-agents to Patch version 1.6.1

# -----BEGIN PGP SIGNATURE-----
#
# iQIzBAABCgAdFiEEPRP5d5L03rt/wvKse019lts08b8FAmkeDnAACgkQe019lts0
# 8b+3mRAAvehfZs7NoAdzWzHQCiYWZlbJ5NYIXPJ2GodktbEQ24xVVPWWlD+uY/Ay
# UM5mm/VroUXlA9f314eC6PwWLkk20uIU+w7jEnJNACzd5PHd5W0H3Gx2uM2B8ZeS
# ifhvJqtXXlBDBgOtoD+oVosB1gTZoUI1uAnW+PcDIU45vs0/QhDPKgFYrOwWklUZ
# m38jZv3gnQ2eMNqQlK/Ohxci/v96Qt8IFLjqcmBBgt5xQqsZeJyhYzno1kY0F18N
# epVjZsVJRt62x8JRCVB4fNM12wIbRPydZhira0Xkxk0kNBqpyGPzuJ0t/xYsjTSu
# pCYbkypjQhmzpVJ7zNN3nUHoB6nCnl1NNWMc8anhyd/XwZo+SZvnfq3Uwb7co714
# HmOlAq3Luv5LC5zh8AzJQb9Uvryg/QHjbR8UZpD/9v4AoLrtaOYjFI8R3JA/eJ+r
# McQCCD7oC7HwuLdRhX5ZXa55Ry9ZqyRIFTHrPKPQtajD9fZVfk1tRZnJCESW4xgK
# PpvuWEUrEIgmm5pYB3UwsFF64LFBe/+KFSl4igebgR7qszhQIJgti4FVbDp7O3EC
# +VAq4IdOWTbqwg6M/TlqTKxOWL53DxLm0/asUI3GDOYF/xYGIYegPcr2nOqSD/9C
# QWNGqPEQ7HhC9WBPMtnd0UBI64rEeGQn6dsETsfbv5d2jVgedaU=
# =luQL
# -----END PGP SIGNATURE-----
# gpg: Signature made mié 19 nov 2025 19:37:36 CET
# gpg:                using RSA key 3D13F97792F4DEBB7FC2F2AC7B4D7D96DB34F1BF
# gpg: Can't check signature: No public key

nickmisasi committed:
Fix migration always reporting wasUpdated=true on empty config (#455)

When there were no bots AND no old services to migrate, migrateServicesToBots()
was incorrectly returning true, causing unnecessary config saves and potential
infinite OnConfigurationChange loops.

Added early return when old services array is empty.

agarciamontoro committed:
Merge tag 'v1.6.2' into CLD-9438-build-agents-fips-complaint

Bumping mattermost-plugin-agents to Patch version 1.6.2

agarciamontoro committed:
Trigger CI

agarciamontoro committed:
Merge tag 'v1.7.0' into CLD-9438-build-agents-fips-complaint

Bumping mattermost-plugin-agents to Minor version 1.7.0

agarciamontoro committed:
Merge tag 'v1.7.1' into CLD-9438-build-agents-fips-complaint

Bumping mattermost-plugin-agents to Patch version 1.7.1

# -----BEGIN PGP SIGNATURE-----
#
# iQIzBAABCgAdFiEEPRP5d5L03rt/wvKse019lts08b8FAmk68X8ACgkQe019lts0
# 8b+TWRAAsyOmWx6ut89xQWq98HYAFBSLxy+rPD/WRd3tG1HCLshzL8htQYnalySV
# fQqwKgHMoThKR66EksXKt7fXqr35BYC3xMOaa1a100n6jS58FPxHVPw6UrGYiu1J
# 99prtnekpB85Pv0TjbOeRmqpFBqXavtIDsGl/5/lFaKYkQlBwCHhX48OCLZOUc8x
# SRcSp0LZF3j3eEGXEBXXDcK+S9M++Tsd6zQQAwSdECViSa3Xysf3+mOYIMADi5Qj
# wCQ0OYENa40ZoZDjtEUu1XQwTQiIyi/Fm3qejvL1wAMNIeagLPM0P80397aEtfxm
# xi/JHAPTLfVmwzFMiO0RHj+cQj2nkmPJ8vCoUcmiHi24UUcvjrVYhG5sBiUIUAYD
# kgNUNu+wifFmr+a1JPEoJ8La3hZav2tkMtUIqsYzDdvyBSQb8p5/xt6eq/Ejyhd9
# WnvStFJ9UQ9ix1GSlI2CiO9eYNPYbxUqs6RGh7fOtfWwNskUNYLtB5y/i75xKJfv
# urdp6zcwFB3GpOGvJbw4GQIKmcL2Ucn385RdkTjO4UucW3TsoLEDk+Us0q3w7NW+
# lXxBDLBKAhjtCLILa5iVqEpI7HKdPpQRD6zi0Gjo34r3yV90LHQlH7han3+ZypRD
# W0eQE8yErXOO/GCRBvC9P5AyFrQHGy9Mr1hAtyOy/rR5PJ13oLk=
# =JPmp
# -----END PGP SIGNATURE-----
# gpg: Signature made jue 11 dic 2025 17:29:51 CET
# gpg:                using RSA key 3D13F97792F4DEBB7FC2F2AC7B4D7D96DB34F1BF
# gpg: Can't check signature: No public key

agarciamontoro committed:
Merge tag 'v1.7.2' into CLD-9438-build-agents-fips-complaint

agarciamontoro committed:
Merge tag 'v2.0.0' into CLD-9438-build-agents-fips-complaint

Bumping mattermost-plugin-agents to Major version 2.0.0

agarciamontoro committed:
Update Go version to match go.mod

agarciamontoro committed:
Bump FIPS image to 1.26.1 to match go.mod

agarciamontoro committed:
Merge tag 'v2.0.2' into CLD-9438-build-agents-fips-complaint

agarciamontoro committed:
Bump FIPS image to 1.26.2 to match go.mod

nickmisasi committed:
Revert "MM-67913: stop toggling app__body on Agents product page (#683)" (#704)

This reverts commit 685c20ef7afcd6025a367cede4343a85a6813987.

agarciamontoro committed:
Merge tag 'v2.0.3' into CLD-9438-build-agents-fips-complaint
